### ISSUE TITLE
connections/http.rb failing on next

### ISSUE THREAD

**Issue by gchpaco** [rethinkdb/rethinkdb#3626]
When running `./test-runner connections/http.rb` I'm seeing these errors.  I'm doing this in a virtualenv, with the following package lists:
* cffi (0.8.6)
* characteristic (14.3.0)
* cryptography (0.7.2)
* enum34 (1.0.4)
* gnureadline (6.3.3)
* ipython (2.3.1)
* pip (1.5.6)
* pyasn1 (0.1.7)
* pyasn1-modules (0.0.5)
* pycparser (2.10)
* pyOpenSSL (0.14)
* PyYAML (3.11)
* service-identity (14.0.0)
* setuptools (3.6)
* six (1.9.0)
* Twisted (14.0.2)
* wsgiref (0.1.2)
* zope.interface (4.1.2)

The Python version is 2.7.9, and the error is as follows:
```
$ ./test-runner connections/http.rb
Warning: did not find any tests for javascript
Warning: did not find any tests for python 2.6
Using rethinkdb binary /Users/graham/wd/rethinkdb/build/debug_clang/rethinkdb
	ruby 2.0 interpreter: /usr/bin/ruby, driver: /Users/graham/wd/rethinkdb/drivers/ruby
== Starting: connections/http.rb2.0 (T+ 0.7 sec)
>>> Failed connections/http.rb2.0 with exit code 1 after 1.4 sec (T+ 2.2 sec)

/Users/graham/wd/rethinkdb/test/rql_test/connections/http.rb:39:in `expect_error': Expected an error, but got success with result:  (RuntimeError)
	from /Users/graham/wd/rethinkdb/test/rql_test/connections/http.rb:226:in `test_part'
	from /Users/graham/wd/rethinkdb/test/rql_test/connections/http.rb:234:in `test_verify'
	from /Users/graham/wd/rethinkdb/test/rql_test/connections/http.rb:269:in `call'
	from /Users/graham/wd/rethinkdb/test/rql_test/connections/http.rb:269:in `block in <main>'
	from /Users/graham/wd/rethinkdb/test/rql_test/connections/http.rb:267:in `each'
	from /Users/graham/wd/rethinkdb/test/rql_test/connections/http.rb:267:in `<main>'
RethinkDB Driver Port: 53451
Running test test_get
 - PASS
Running test test_params
 - PASS
Running test test_headers
 - PASS
Running test test_head
 - PASS
Running test test_post
 - PASS
Running test test_put
 - PASS
Running test test_patch
 - PASS
Running test test_delete
 - PASS
Running test test_redirects
 - PASS
Running test test_gzip
 - PASS
Running test test_failed_json_parse
 - PASS
Running test test_basic_auth
 - PASS
Running test test_digest_auth
 - PASS
Running test test_verify

<<< end Failed: connections/http.rb2.0

== Failed 1 test (of 1) in 2.34 seconds!
	connections/http.rb2.0
```
It appears to be that the problem is the peer cert isn't being bounced.  Given that this is a fresh virtualenv with fresh pyOpenSSL and Twisted I think there may be something here.

**Comment by danielmewes** [comment]
@larkost do you have seen this before or have an idea?

**Comment by Tryneus** [comment]
Is this on OSX?  Do the `http.py` and `http.js` tests fail as well?  Which SSL library is your `rethinkdb` executable linked against?  It's possible this might be solved by #3540.

**Comment by gchpaco** [comment]
This is on OS X, yes.  It could well be connected to #3540.  `http.js` is failing, `http.py` is not.  The `rethinkdb` executable is linked to `/usr/lib/libcrypto.0.9.8.dylib`, as is the Python cryptography libraries.

**Comment by gchpaco** [comment]
(I should say `http.js` is failing for the same reason)

**Comment by Tryneus** [comment]
I should mention that my previous question about the SSL library is irrelevant, what is important is how `rethinkdb` is linking against `libcurl` and which SSL library `libcurl` is using.

**Comment by gchpaco** [comment]
I don't know, but `otool -L` computes the transitive closure of libraries linked against, and `libcurl` didn't show up on the list implying that it is statically linked (not unusual, I run in `--allow-fetch`).  In that case the only SSL related thing linked is `libcrypto`.

**Comment by danielmewes** [comment]
@gchpaco if you fetch libcurl during the build, you could manually edit the first line of `mk/support/pkg/curl.sh` to read `version=7.40.0`. That way you can test if https://github.com/rethinkdb/rethinkdb/issues/3540 is going to solve the issue.

**Comment by larkost** [comment]
And we never understood why we were not seeing the issue on our statically compiled versions, only observed that we were not. And from my reading of the fix, we should be.

**Comment by larkost** [comment]
I have not seen this issue since we bumped the curl version to newer than the fix in #3540. OS closing.
